### DISCUSSION TITLE
Customize Extensions in WebSocketGraphQLClient reactiveExecuteQuery request

### DISCUSSION THREAD

**Discussion by anderson-souza-involves** [Netflix/dgs-framework#1673] [Q&A] [answered]
Hello folks,

I have a question, is it possible to customize the `extensions` property from the `QueryPayload` while sending a `subscription` in the `WebSocketGraphQLClient.reactiveExecuteQuery`?

I'm trying to create a `subscription` request for a GraphQL hosted on AWS AppSync. In their [documentation](https://docs.aws.amazon.com/appsync/latest/devguide/real-time-websocket-client.html#subscription-registration-message), it says that I need to send in the `extensions` property an Object containing the Authorization header, but I haven't found a way to add values to this property while using the client.

The method `reactiveExecuteQuery` don't allow me to pass any data to the `extensions` property.

Is there any other way I can manage this `extensions`?

Thanks in advance!

**Comment by srinivasankavitha** [comment, marked as answer]
We don't have one currently. We will need to add a new method in the interface that allows you to set extensions as part of the request. Will try to add that in an upcoming release.

**Reply by anderson-souza-involves** [reply]
Thanks, I'll keep an eye out for the next releases :)
